### PR TITLE
Use OpenAPI generated API

### PR DESCRIPTION
**Pull request by dpordomingo**
caused by https://github.com/athenianco/athenian-webapp/pull/23
blocks https://github.com/athenianco/athenian-webapp/pull/25

This PR adds OpenAPI Generator, that will generate the API from `athenian-api/openapi.yml` specs.
The autogenerated API won't be committed nor pushed.
The server must be now run with:
```bash
$ make serve
```

This adds no other feature, nor performs any other refactor.

### TODO (in further PR's)

- Refactor the API with generic code, error handlers, toasts, separation between data access and data mapping...

### CODE REVIEW

**Comment by se7entyse7en** [comment]
> The autogenerated API won't be committed nor pushed.

I don't understand the reason behind this.

File: Makefile [hunk 1]
```diff
@@ -8,3 +8,49 @@ MAKEFILE := $(CI_PATH)/Makefile.main
 $(MAKEFILE):
 	git clone --quiet --depth 1 -b $(CI_BRANCH) $(CI_REPOSITORY) $(CI_PATH);
 -include $(MAKEFILE)
+
+# If CI flags is set, build will fail if there are any WARNING log, and tests will run and exit instead of watching for changes.
+# CI is set to '1' in travis by default
+CI ?=
+
+# TODO (dpordomingo): consider locking to fixed version once we start doing releases.
```
Comment (se7entyse7en) [inline Makefile]: Good point

File: Makefile [hunk 2]
```diff
@@ -8,3 +8,49 @@ MAKEFILE := $(CI_PATH)/Makefile.main
...
+
+.PHONY: clean
+clean: clean-openapi
+
+.PHONY: clean-openapi
+clean-openapi:
+	rm -rf $(API_CLIENT_CODE_PATH)
+
+.PHONY: build
+build: generate
```
Comment (se7entyse7en) [inline Makefile]: We need to change `Dockerfile` as well.

File: src/services/api/index.js [hunk 3]
```diff
@@ -88,47 +93,42 @@ const pipeline = [
 ];
 
 const api = metrics => {
-  const repos = [
+
+  window.DefaultApi = DefaultApi;
+  window.MetricsRequest = MetricsRequest;
```
Comment (se7entyse7en) [inline src/services/api/index.js]: Why you put on `window`?

Comment (dpordomingo) [reply src/services/api/index.js]: good catch. This was a leftover.

File: Makefile (hunk 2 above)
Comment (dpordomingo) [reply Makefile]: done by e7a0504

**Comment by dpordomingo** [comment]
Autogenerated OpenAPI client was commited and pushed by 83f6cc4d5f6b0c26ca82a93b07ca42cbcefa0144

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -24,6 +24,10 @@ This project was bootstrapped with [Create React App](https://github.com/faceboo
...
-- `yarn install`: Install webapp dependencies for the very first time,
-- `yarn start`: Runs the webapp in the development mode under http://localhost:3000. The page will hot-reload on every edits, and lint errors will appear in the console,
-- `yarn test`: Runs the app tests in interactive watch mode,
+- `make serve`: Runs the webapp in the development mode under http://localhost:3000. The page will hot-reload on every edits, and lint errors will appear in the console,
+- `make test`: Runs the app tests in interactive watch mode,
+
+#### Autogenerated code
+
+The following commands auto generate different parts of the app:
+- `make serve`: Generates the OpenAPI client
```
Comment (se7entyse7en) [inline docs/CONTRIBUTING.md]: I guess you mean
```suggestion
- `make generate`: Generates the OpenAPI client
```

**Comment by dpordomingo** [comment]
merged by https://github.com/athenianco/athenian-webapp/pull/25

**Pull request merged**